jigglypuffRL/common/utils.py

```diff
@@ -0,0 +1,26 @@
+import torch
+
+
+def save_params(algo, name=None, version=None):
+    if name is None:
+        name = algo.network_type
+
+    checkpoint = {'algo': name,
+                  'weights': algo.ac.state_dict()}
+
+    if version is None:
+        torch.save(checkpoint, "{}.pt".format(name))
+    else:
+        torch.save(checkpoint, "{}-{}.pt".format(name, version))
+    return checkpoint
+
+
+def load_params(name, version=None):
+    try:
+        if version is None:
+            checkpoint = torch.load("{}.pt").format(name)
+        else:
+            checkpoint = torch.load("{}-{}.pt".format(name, version))
+    except FileNotFoundError:
+        raise Exception("Check name and version number again")
+    return checkpoint

```

Comment:
Also, can you update the object params according to the new save function. I think you can do something like loop over the dictionary items and use `setattr`. 

The model weights will have to be loaded in a separate statement still though

Reply:
Changed it. Model should be exactly reproduced (didn't use deepcopy for the model so I'm not sure if it's "exactly") with all parameters as I used algo.\_\_dict\_\_